### ISSUE TITLE
Storybook 4.0.0-alpha.1 viewport-addon works only as panel on angular and the javascript Api doesn't.

### ISSUE THREAD

**Issue by gpincheiraa** [storybooks/storybook#3321]
I'm trying to use the version 4.0.0-alpha.1 with angular and works really good. The problem it is when I tried to use the Javascript API for manipulate the addon-viewport. As a panel the addon works good but when I tried to set the viewport for some stories I get the following error:

```
Warning: Failed prop type: Invalid prop `children` supplied to `Viewport`, expected a ReactNode.
    in Viewport
```
and
```
Unhandled Promise rejection: Unexpected value 'undefined' declared by the module 'DynamicModule' ; Zone: <root> ; Task: Promise.then ; Value: Error: Unexpected value 'undefined' declared by the module 'DynamicModule'
    at syntaxError (compiler.js:485)
    at compiler.js:15273
    at Array.forEach (<anonymous>)
    at CompileMetadataResolver../node_modules/@angular/compiler/esm5/compiler.js.CompileMetadataResolver.getNgModuleMetadata (compiler.js:15271)
```

My code it is:

**.storybook/config.js**
```
import { configure } from '@storybook/angular';

function loadStories() {
	require('../stories/index.ts');
}
configure(loadStories, module);
```

**.storybook/addons.js**
```
import '@storybook/addon-viewport/register';
```
**stories/myStory.js**
```
import { storiesOf, addDecorator } from '@storybook/angular';
import { withViewport } from '@storybook/addon-viewport';
import { HeaderComponent } from '../../src/app/modules/ui-common/header/header.component';

storiesOf('Header component Mobile', module)
	.addDecorator(withViewport('iphone6'))
	.add('Header white', () => ({
		component: HeaderComponent
	}));
```
My dependencies Installed:

**npm install --depth=0**
```
Dependencies:
├── @angular/animations@5.2.4
├── @angular/cli@1.6.8
├── @angular/common@5.2.4
├── @angular/compiler@5.2.4
├── @angular/compiler-cli@5.2.4
├── @angular/core@5.2.4
├── @angular/forms@5.2.4
├── @angular/http@5.2.4
├── @angular/language-service@5.2.4
├── @angular/platform-browser@5.2.4
├── @angular/platform-browser-dynamic@5.2.4
├── @angular/router@5.2.4
├── @storybook/addon-viewport@4.0.0-alpha.1
├── @storybook/addons@4.0.0-alpha.1
├── @storybook/angular@4.0.0-alpha.1
├── @storybook/storybook-deployer@2.3.0
├── @types/jest@22.2.2
├── @types/node@6.0.101
├── blink-diff@1.0.13
├── codelyzer@4.1.0
├── concurrently@3.5.1
├── core-js@2.5.3
├── cypress@2.1.0
├── cypress-cucumber-preprocessor@0.3.0
├── jest@22.4.3
├── jest-preset-angular@5.2.1
├── rxjs@5.5.6
├── ts-node@4.1.0
├── tslint@5.9.1
├── typescript@2.5.3
├── wait-on@2.1.0
└── zone.js@0.8.20
```
Edit: You can reproduce this error on the following repo and branch:
https://github.com/gpincheiraa/storybook-components-sample/tree/addons-viewport

**Comment by mshaaban088** [comment]
Thanks @gpincheiraa for reporting this issue.
Unfortunately the viewport's `decorators` (a.k.a APIs) only work for `React`, as it uses `React` wrapper component internally, I should have mentioned that in the documentation though.

I believe this functionality should be supported in other libraries/frameworks (e.g. angular, vue), probably I'm gonna work on it by next week.

**Comment by mshaaban088** [comment]
I've tested it now with [releases/v4.0.0-alpha.7](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.7), and it works pretty well

**Comment by Hypnosphi** [comment]
Yes, this was probably fixed with #3555
